### ISSUE TITLE
Recommendation on LR and other hyper params on `ketos segtrain` as well as what metric to use

### ISSUE THREAD

**Issue by PonteIneptique** [mittagessen/kraken#262]
Hi @mittagessen ;)
We have been running the following command with our data (a lot of different region and lines, and we have a unified one with --merge-regions and --merge-baselines): `ketos segtrain -bl -f alto -o Model --device cuda:0 --threads 10 --augment --epochs 50 --schedule reduceonplateau  -s "[1,1800,0,3 Cr7,7,64,2,2 Gn32 Cr3,3,128,2,2 Gn32 Cr3,3,128 Gn32 Cr3,3,256 Gn32 Cr3,3,256 Gn32 Lbx32 Lby32 Cr1,1,32 Gn32 Lby32 Lbx32]" ` 

1. Regarding LR and other hyperparams, do you have any recommendation ?
2. When the training is done, what is the metric you think we should focus on to chose the model ?

Thanks in advance !

**Comment by mittagessen** [comment]
1. The default learning rate seems to work quite well, although you can turn it down a notch if you reduce/disable the weight decay. ReduceOnPlateau gets you reliably a bit more accuracy out of recognition models but as the cross entropy loss of the segmentation network is only indirectly linked to the actual quality of the segmentation it's a bit more tricky to say if it actually helps. At least it doesn't seem to hurt.
2. That depends what you're training for. Because of the required postprocessing the raw pixel-based scores do not mean that much for the baselines while the IoU scores for regions can be directly interpreted. Generally, higher scores are better. In addition, we've seen that running the training for a few more epochs when the scores are just oscillating around some value seems to help with generalization. Building a good metric for validation purposes is on the todo list, albeit very low priority right now.

**Comment by PonteIneptique** [comment]
Thanks a lot for the feedback :)

Le lun. 28 juin 2021 à 20:25, mittagessen ***@***.***> a
écrit :
